admin/includes/init_includes/init_special_funcs.php

```diff
@@ -6,29 +6,53 @@
  * @version $Id: DrByte 2020 Jul 10 Modified in v1.5.8-alpha $
  */
 if (!defined('IS_ADMIN_FLAG')) {
-  die('Illegal Access');
+    die('Illegal Access');
 }
+
 // set a default time limit
-  zen_set_time_limit(GLOBAL_SET_TIME_LIMIT);
+zen_set_time_limit(GLOBAL_SET_TIME_LIMIT);
 
 // auto activate and expire banners
-  require(DIR_FS_CATALOG . DIR_WS_FUNCTIONS . 'banner.php');
-  zen_activate_banners();
-  zen_expire_banners();
-
-// auto expire special products
-  require(DIR_FS_CATALOG . DIR_WS_FUNCTIONS . 'specials.php');
-  zen_start_specials();
-  zen_expire_specials();
-
-// auto expire featured products
-  require(DIR_FS_CATALOG . DIR_WS_FUNCTIONS . 'featured.php');
-  zen_start_featured();
-  zen_expire_featured();
-
-// auto expire salemaker sales
-  require(DIR_FS_CATALOG . DIR_WS_FUNCTIONS . 'salemaker.php');
-  zen_start_salemaker();
-  zen_expire_salemaker();
-
-?>
\ No newline at end of file
+require DIR_FS_CATALOG . DIR_WS_FUNCTIONS . 'banner.php';
+zen_activate_banners();
+zen_expire_banners();
+
+// -----
+// Functions are still needed for specials/featured/salemaker tools'
+// processing!
+//
+require DIR_FS_CATALOG . DIR_WS_FUNCTIONS . 'specials.php';
+require DIR_FS_CATALOG . DIR_WS_FUNCTIONS . 'featured.php';
+require DIR_FS_CATALOG . DIR_WS_FUNCTIONS . 'salemaker.php';
```

Comment:
I'm inclined to relocate these up above the expiring of banners: ie: include/require all the files ... cuz this init file is tasked with loading those functions files.
And then do all the activating/expiring/etc of banners and specials/etc in the last half of the file.

I'm not "bent" on this suggestion. It was your "are still needed" comment that led me to consider "what do you mean by 'still needed'?" ... cuz those files must always get loaded for core code to work too, not just for expiring/etc.

Reply:
My "still needed" comment was after-the-fact (on the client's update) realizing that the `require` of the functions needed to be outside of the does-this-need-to-be-run section, since the Specials/SaleMaker tools use the functions therein to 'toggle' an elements status.